gravitee-apim-console-webui/src/management/api/documentation-v4/components/api-documentation-choose-page-list/api-documentation-choose-page-list.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { MatTreeModule } from '@angular/material/tree';
+import { MatRadioModule } from '@angular/material/radio';
+import { MatIconModule } from '@angular/material/icon';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import { FormsModule } from '@angular/forms';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+
+import { ApiDocumentationChoosePageListComponent, PageData } from './api-documentation-choose-page-list.component';
+
+import { Page } from '../../../../../entities/management-api-v2';
+
+describe('ApiDocumentationChoosePageListComponent', () => {
+  let component: ApiDocumentationChoosePageListComponent;
+  let fixture: ComponentFixture<ApiDocumentationChoosePageListComponent>;
+
+  const pages: Page[] = [
+    { id: 'page-id', name: 'Page 1', type: 'MARKDOWN', published: true, visibility: 'PUBLIC', generalConditions: false, parentId: null },
+    {
+      id: 'folder-id',
+      name: 'Folder 1',
+      type: 'FOLDER',
+      published: false,
+      visibility: 'PRIVATE',
+      generalConditions: false,
+      parentId: null,
+    },
+  ];
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [
+        NoopAnimationsModule,
+        MatTreeModule,
+        MatRadioModule,
+        MatIconModule,
+        MatTooltipModule,
+        FormsModule,
+        ApiDocumentationChoosePageListComponent,
+      ],
+      providers: [PageData],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ApiDocumentationChoosePageListComponent);
+    component = fixture.componentInstance;
+    component.pages = pages;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should render header with text "Name"', () => {
+    const headerElement = fixture.debugElement.query(By.css('.header'));
+    expect(headerElement).toBeTruthy();
+    expect(headerElement.nativeElement.textContent).toContain('Name');
+  });
+
+  it('should render mat-tree', () => {
```

Comment:
Instead of using technical language (`mat-tree`), it should speak to the business rules of the component. The same goes for the tests below.

```suggestion
  it('should render the page list', () => {
```

Reply:
DOne